### DISCUSSION TITLE
Installation/Usage tutorial? [Synology & Docker]

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [cross-seed/cross-seed#244] [Q&A] [answered]
Hey, all.

I just found this script, but I'm having trouble understanding what exactly I need to do in order to install and use this. I use Docker in Synology, but I don't know how to work with docker CLI since I've always just used their Docker package. I do have intermediate CLI knowledge in general, but I'm not really connecting the dots on what I need to do in order to use this.

I've downloaded the docker image for this, and I have created a /cross-seed/config/ directory, as I do for all docker containers. But I don't know how to configure or run cross-seed at this point. I'm assuming I have to SSH into my Syno to run the CLI... but my end goal is to get qBT to auto run the script upon download completion.

Is there any tutorial or documentation out there for those of us who aren't developers or more advanced users to follow? I just want to make sure I know what I'm doing.

Thanks!

**Comment by mmgoodnow** [comment]
Have you read the wiki? Here's some information on how yo use docker-compose: https://github.com/mmgoodnow/cross-seed/wiki/Docker#configuration

**Comment by mmgoodnow** [comment]
You might have to read up a bit on how to start Docker containers either with or without Docker-compose. 

To configure, you can run the container once and it should create a config file for you which you can then go in and edit. The config file uses JavaScript syntax, so make sure you put double quotes around your strings.

**Comment by SawkeeReemo** [comment]
Yeah, I have several Docker containers already. I use Portainer, etc. But there’s just not enough information in the wiki if you’re not a developer. I’m sure lots of things are second nature to those who code, but for those who are not, like me, I’ve been trying to figure this out for hours but too much information is missing.

This is why I’m asking if there’s a tutorial somewhere showing how to install and use this. Then it would make more sense to me.

**Comment by mmgoodnow** [comment, marked as answer]
Unfortunately the documentation is pretty bare-bones. There is not a tutorial. 

I can help you if you tell me what you're currently stuck on.

**Comment by SawkeeReemo** [comment]
Oh thank you! That’s super cool of you. I’m actually heading into a (potentially long) meeting for work right now, but I’ll definitely take you up on that!

**Comment by SawkeeReemo** [comment]
I got this installed on my Mac instead of my NAS for now so I can at least run these ~~one at a time~~ manually for the time being. But I'm looking over the config file right now and it has conflicting info:

```
    /**
     * List of Torznab urls.
     * The path should end in /api
     * For Jackett, click "Copy RSS feed"
     * For Prowlarr, click (i) and copy the Torznab Url, then append "?apikey=YOUR_PROWLARR_API_KEY"
     */
    torznab: [],
```

When I click "Copy RSS feed" in Jackett per tracker, I don't get a URL that ends in `/api`. Mine all end in my API key then `&t=search&cat=&q=`. So I'm not sure what's going on or if this is going to work. Also, I'm assuming you separate each URL with a comma, right? The config message doesn't specify (or if the URLs need to be in quotes, etc). Also, why would I click "copy RSS feed" when this is supposed to be a torznab feed? This stuff isn't very clear.

**Reply by SawkeeReemo** [reply]
So this just error'd out giving me an "unexpected token" error with that token being a colon. I went back and wrapped the URL's in quotes, and then it worked. Also, the whole `* The path should end in /api` is actually not correct either since my Jackett RSS feed URLs don't end that way and worked once I wrapped them in quotes.

**Reply by mmgoodnow** [reply]
Sounds like you figured it out.

The URL _**path**_ does end in `/api`. All the stuff after the `?` is not part of the path, but what's known as the "query string".

**Reply by SawkeeReemo** [reply]
Good to know. But you can see how that would be confusing to someone setting this up for the first time who doesn’t know that, yeah? I’ve set up and configured a lot of things on my own, self taught stuff mostly as a hobby for many years, but I’ve never come across that. I assumed what I was doing would work, but I had to guess because it was not clear. 

And I got it working just on my Mac. But all my qBT and Sonarr/Radarr stuff is hosted on my NAS. That’s the part I can’t figure out with this, and I think it’s because I can’t seem to figure out how to upgrade Node.js on my Synology running DSM 6.2. I’d love to run the daemon version of this to be executed as a script upon download completion. ~~But also seeing as I don’t think you can tell the script to select a specific category, I’m not sure at the moment how to make this work with my setup even if I can get Node updated.~~

UPDATE: this last sentence is moot when using inject. I've learned a lot since this initial post. Thank you for your patience!
